avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -362,28 +373,75 @@ Object read(Schema writerSchema) {
     }
 
     Object read(Schema writerSchema, Schema readerSchema) {
-      DatumReader<?> reader = getDatumReader(writerSchema, readerSchema);
-      int length = buffer.limit() - 1 - idSize;
-      if (writerSchema.getType().equals(Schema.Type.BYTES)) {
-        byte[] bytes = new byte[length];
-        buffer.get(bytes, 0, length);
-        return bytes;
-      } else {
-        int start = buffer.position() + buffer.arrayOffset();
-        try {
+      try {
+        DatumReader<?> reader = getDatumReader(writerSchema, readerSchema);
+        int length = buffer.limit() - 1 - idSize;
+        if (writerSchema.getType().equals(Schema.Type.BYTES)) {
+          byte[] bytes = new byte[length];
+          buffer.get(bytes, 0, length);
+          return bytes;
+        } else {
+          int start = buffer.position() + buffer.arrayOffset();
           Object result = reader.read(null, decoderFactory.binaryDecoder(buffer.array(),
               start, length, null));
           if (writerSchema.getType().equals(Schema.Type.STRING)) {
             return result.toString();
           } else {
             return result;
           }
-        } catch (IOException | RuntimeException e) {
-          // avro deserialization may throw AvroRuntimeException, NullPointerException, etc
-          throw new SerializationException("Error deserializing Avro message for id "
-              + schemaId, e);
         }
+      } catch (ExecutionException ex) {
+        throw new SerializationException("Error deserializing Avro message for id "
+            + schemaId, ex.getCause());
```

Comment:
Why `ex.getCause` instead of just `ex` ?

Reply:
ExecutionException is a wrapper class used by Guava cache